engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -2756,21 +2756,23 @@ namespace dmGameSystem
                 return dmGameObject::PROPERTY_RESULT_INVALID_KEY;
             }
             dmResource::HFactory factory = dmGameObject::GetFactory(params.m_Instance);
-            dmGameSystem::FontResource* font = 0;
-            dmGameObject::PropertyResult res = SetResourceProperty(factory, params.m_Value, FONT_EXT_HASH, (void**)&font);
+            dmGameSystem::FontResource* font_resource = 0;
+            dmGameObject::PropertyResult res = SetResourceProperty(factory, params.m_Value, FONT_EXT_HASH, (void**)&font_resource);
             if (res == dmGameObject::PROPERTY_RESULT_OK)
             {
+                dmRender::HFont font = dmGameSystem::ResFontGetHandle(font_resource);
```

Comment:
The bug fix:
We recently updated from a HFont to a FontResource* as the actual resource pointer.
And I had missed this case. The gui system stores pointers to the HFont, whereas I accidentally passed in a FontResource*, making the types mismatched, and generated an invalid access.